### PR TITLE
feat(server): thread and comment schema

### PR DESCRIPTION
**Pull request by nourbalaha**
# Overview
In this PR I added a schema for thread and comment.

### CODE REVIEW

File: server/schemas/thread.graphql [hunk 1]
```diff
@@ -0,0 +1,35 @@
+type Thread {
+  id: ID!
+  comments: [Comment]
```
Comment (rot1024) [inline server/schemas/thread.graphql:3]:
```suggestion
  workspaceId: ID!
  comments: [Comments!]!
  workspace: Workspace
```

workspace should be a "resolver"

File: server/schemas/thread.graphql [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  content: String!
+}
+
+input CreateCommentInput {
+  threadId: ID!
+  authorId: ID!
+  content: string!
+}
+
+input UpdateCommentInput {
```
Comment (rot1024) [inline server/schemas/thread.graphql:18]:
```suggestion
input UpdateCommentInput {
  threadId: ID!
```

File: server/schemas/thread.graphql [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  commentId: ID!
+  content: string!
+}
+
+input DeleteCommentInput {
+  CommentId: ID!
```
Comment (rot1024) [inline server/schemas/thread.graphql:24]:
```suggestion
  threadId: ID!
  commentId: ID!
```

File: server/schemas/thread.graphql [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+extend type Query {
+  comments(threadId: ID!): [Comment]
+}
```
Comment (rot1024) [inline server/schemas/thread.graphql:29]:
```suggestion
```

I think it will not be used

File: server/schemas/thread.graphql [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+extend type Mutation {
+  createComment(input: CreateCommentInput!): Comment: Comment!
+  UpdateComment(input: UpdateCommentInput!): Comment: Comment!
```
Comment (rot1024) [inline server/schemas/thread.graphql:33]:
```suggestion
  updateComment(input: UpdateCommentInput!): Comment: Comment!
```

File: server/schemas/thread.graphql [hunk 6]
```diff
@@ -0,0 +1,35 @@
+type Thread {
+  id: ID!
+  comments: [Comment]
+}
+
+type Comment {
+  id: ID!
+  authorId: ID!
+  content: String!
```
Comment (rot1024) [inline server/schemas/thread.graphql:9]:
```suggestion
  content: String!
  createdAt: DateTime!
```

createdAt can use a timestamp of a commnet ID, so there is no need to add a `createdAt` field to the domain model.

**Review by rot1024 — CHANGES_REQUESTED** [review]
thread.graphql -> comment.graphql

File: server/schemas/thread.graphql [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+  id: ID!
+  author: User
+  authorId: ID!
+  content: String!
+  createdAt: DateTime!
+}
+
+input CreateCommentInput {
+  threadId: ID!
+  authorId: ID!
```
Comment (yk-eukarya) [inline server/schemas/thread.graphql:18]: Do we need the `authorId` here? cant we use the logged-in user id from `operator`

Comment (nourbalaha) [reply server/schemas/thread.graphql:18]: @yk-eukarya I think that we do not need authorId here, this applies to `UpdateCommentInput` as well right?

Comment (nourbalaha) [reply server/schemas/thread.graphql]: Done!

**Review by rot1024 — APPROVED** [review]

**Pull request merged**